modules/mill-codegen-plugin/src/smithy4s/codegen/mill/Smithy4sModule.scala

```diff
@@ -74,7 +74,7 @@ trait Smithy4sModule extends ScalaModule {
   def smithy4sCodegen: T[(PathRef, PathRef)] = T {
 
     val specFiles = if (os.exists(smithy4sInputDir().path)) {
-      os.walk(smithy4sInputDir().path, skip = _.ext != "smithy")
+      os.list(smithy4sInputDir().path)
```

Comment:
why list at all ? What happens if we just pass the folder ?

Reply:
if we just pass the folder, we need extra work to strip `smithy` from the directory structure.

You could argue it's better because we move more stuff out of the plugins into the shared code (`SmithyResources`)...

pushed a commit that does that, lemme know if it looks better. I also added a `.filter(os.isFile(_))` in the `os.walk` call so that we never list directories in `smithy/manifest`.